### ISSUE TITLE
Solve ambiguity for deNovoPrior = 0.02 in ISPP Sandbox

### ISSUE THREAD

**Issue by amycoffin** [BRCAChallenge/brca-exchange#818]
splicePrior can equal 0.02 for more than one reason:

- The de novo splice likelihood is weak/null, or low. 
- The de novo splice site creates an innocuous In Frame Deletion (IFD). It is also ambiguous (to me) whether this IFD is the same IFD noted in issue #816  

This ambiguity affects the Splice-level estimation nested tile, the de novo donor tab.

**Comment by amycoffin** [comment]
according to calcVarPriors.py, "frameshiftFlag: equals 1 if variant causes a frameshift, 0 if not". However, there are also "-" and "N/A" values assigned for that flag. It appears that all insertions, deletions, and indels are currently "-", which means that frameshiftFlag might be more useful once Zack implements the indel side of the code. 

Intuitively, I believe that the part of the code that assigns frameshiftFlag should be this simple:
if varType = substitution, then frameShiftFlag = 0
if varType == insertion, deletion, or indel **and** isDivisible == False, then frameshiftFlag = 1

frameshiftFlag might need to be assigned as N/A for the weird exception cases. 

calcVarPriors.py appears to currently handle the frameshiftFlag on a case-by-case basis, while the function "getDeNovoSpliceFrameshiftStatus" provides a T/F on frameshift. Can we simply put the frameshiftFlag at the end fo that function? The only problem is that the frameshift flag is embedded throughout the code in conditionals, and I assume it plays some important role if it shows up 86 times.

**Comment by amycoffin** [comment]
Once issues #831 and #832 are addressed, we can implement a frameshift flag either using getVarType() or getVarConsequnces(), or perhaps a combination of the two.

**Comment by amycoffin** [comment]
Once issues #831 and #832 are addressed, we can implement a frameshift flag either using getVarType() or getVarConsequnces(), or perhaps a combination of the two.

**Comment by amycoffin** [comment]
@falquaddoomi has this been resolved, without updates to the frameshift flag?

**Comment by amycoffin** [comment]
@melissacline is it true that for all substitution variants, an IFD is not possible? So, given the current data release, can we default to Weak/Null because we have not implemented indels yet?

**Comment by melissacline** [comment]
@amycoffin I think we're safe in saying that if a variant is a substitution variant, then it's not an in-frame deletion.  What is it again that you'd like to default to Weak/Null?

**Comment by melissacline** [comment]
@falquaddoomi and @zfisch, when I've looked at this issue with fresh eyes, I've decided that if hte splicing prior is really, really low, then most users are not going to care _which_ situation gave rise to the low prior.  I doubt this will be an issue for this release, since the de novo splice site with the innocuous IFD is a rare situation.  For subsequent releases, it's okay with me if we merge the two boxes, e.g. "Weak/Null & Low OR Innocuous IFD".  That's one option, if it simplifies life.

**Comment by zfisch** [comment]
@melissacline @falquaddoomi @amycoffin what's our status on this issue?

**Comment by amycoffin** [comment]
@falquaddoomi @zfisch last we discussed, this was an okay bug for now, but are we working on getting these ambiguities finally resolved?

**Comment by falquaddoomi** [comment]
@zfisch No, that should do it. I'll merge the two categories ASAP. I presume the same applies to #816.
